### PR TITLE
Notification policy and named webhooks

### PR DESCRIPTION
**Pull request by tomasmik**
## Description of the change

Will stay here until Frontend is merged, but a review before that won't hurt.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (non-breaking change that adds documentation)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development
- [ ] Examples for new resources and data sources have been added
- [ ] Default values have been documented in the description (e.g., "Dummy: (Boolean) Blah blah. Defaults to `false`.)
- [x] If the action fails that checks the documentation: Run `go generate` to make sure the docs are up to date

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Pull Request is no longer marked as "draft"
- [ ] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer
- [ ] The target branch is `future` unless the change is going directly into production

### CODE REVIEW

**Comment by marcinwyszynski** [comment]
Task linked: [CU-2xhx341 [Notification policy] Add policy and new webhooks to TF provider](https://app.clickup.com/t/2xhx341)

File: spacelift/data_named_webhook.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+func dataNamedWebhookRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var query struct {
+		Webhook *structs.NamedWebhooksIntegration `graphql:"namedWebhooksIntegration(id: $id)"`
+	}
+
+	webhookID := d.Get("webhook_id").(string)
+	variables := map[string]interface{}{"id": toID(webhookID)}
+	if err := meta.(*internal.Client).Query(ctx, "GetOneWebhook", &query, variables); err != nil {
```
Comment (michieldewilde) [inline spacelift/data_named_webhook.go:73]: GetNamedWebhook might be better

File: spacelift/data_named_webhook.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		return diag.Errorf("could not query for named webhook: %v", err)
+	}
+
+	if query.Webhook == nil {
+		d.SetId("")
```
Comment (michieldewilde) [inline spacelift/data_named_webhook.go:78]: You want to throw an error when you can't find it.
`return diag.Errorf("could not find named webhook")`

File: spacelift/resource_named_webhook.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "enables or disables sending webhooks. Defaults to `true`.",
+				Optional:    true,
+				Default:     true,
```
Comment (michieldewilde) [inline spacelift/resource_named_webhook.go:35]: Please do not set the default for an optional true, we can not check if the attribute was set by the provider (as the default) or through a field. This causes problems when we might want to change this field and the only way to do the change would be a breaking change.

File: spacelift/resource_named_webhook.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+	}
+
+	variables := map[string]interface{}{"input": input}
+
+	if err := meta.(*internal.Client).Mutate(ctx, "NamedWebhookCreate", &mutation, variables); err != nil {
+		return diag.Errorf("could not create named webhook: %v", internal.FromSpaceliftError(err))
+	}
+
+	if !mutation.WebhooksIntegration.Enabled {
+		return diag.Errorf("named webhook not activated")
```
Comment (michieldewilde) [inline spacelift/resource_named_webhook.go:104]: do we not allow a created webhook to be disabled?

File: spacelift/resource_named_webhook.go [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+	}
+
+	d.SetId(mutation.WebhooksIntegration.ID)
+
+	return nil
```
Comment (michieldewilde) [inline spacelift/resource_named_webhook.go:109]: call your read method here

File: spacelift/resource_named_webhook.go [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+}
+
+func resourceNamedWebhookRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var query struct {
+		Webhook *structs.NamedWebhooksIntegration `graphql:"namedWebhooksIntegration(id: $id)"`
+	}
+
+	variables := map[string]interface{}{"id": graphql.ID(d.Id())}
+	if err := meta.(*internal.Client).Query(ctx, "GetOneWebhook", &query, variables); err != nil {
```
Comment (michieldewilde) [inline spacelift/resource_named_webhook.go:118]: GetNamedWebhook might be better.

File: spacelift/resource_named_webhook_test.go [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+	t.Run("attach a webhook to root space", func(t *testing.T) {
+		config := func(endpoint string) string {
+			return fmt.Sprintf(`
+				resource "spacelift_named_webhook" "test" {
+					endpoint = "%s"
+					space_id = "root"
+					name     = "testing-named-hooks"
+					labels   = ["1", "2"]
+					secret   = "super-secret"
+					enabled  = true
```
Comment (michieldewilde) [inline spacelift/resource_named_webhook_test.go:24]: test defaults

File: spacelift/data_named_webhook.go [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+			},
+			"labels": {
+				Type:        schema.TypeSet,
+				Description: "labels for the webhook to use when referring in policies or filtering them",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Computed: true,
+			},
+			"secret": {
```
Comment (TheMacies) [inline spacelift/data_named_webhook.go:51]: Should we set `Sensitive:true `for this?

**Review by TheMacies — APPROVED** [review]
LGTM with just one comment

**Review by michieldewilde — APPROVED** [review]

**Pull request merged**